addons/account/models/account_move.py

```diff
@@ -1128,7 +1128,21 @@ def _compute_amount(self):
 
     @api.depends('amount_residual', 'move_type', 'state', 'company_id', 'matched_payment_ids.state')
     def _compute_payment_state(self):
-        stored_ids = tuple(self.ids)
+        def _invoice_qualifies(move):
+            currencies = move._get_lines_onchange_currency().currency_id
+            currency = currencies if len(currencies) == 1 else move.company_id.currency_id
+            return move.is_invoice(True) and (move.state == 'posted' or (move.state == 'draft' and not currency.is_zero(move.amount_total)))
```

Comment:
There is only one currency that matters for invoices
```suggestion
            return move.is_invoice(True) and (
                move.state == 'posted' 
                or (move.state == 'draft' and not move.currency_id.is_zero(move.amount_total))
            )
```

Reply:
Should I update [this code](https://github.com/odoo/odoo/pull/212617/files#diff-1e3bd6be3bfb83a37ec9fb800ce8b1c95afe0be90ff792874ae7299c320a2f6eR1181-R1183) as well ?
in this version or master ?